src/test/java/spleetwaise/transaction/model/filterpredicate/CategoryFilterPredicateTest.java

```diff
@@ -0,0 +1,67 @@
+package spleetwaise.transaction.model.filterpredicate;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static spleetwaise.transaction.testutil.TransactionBuilder.DEFAULT_CATEGORY;
+
+import org.junit.jupiter.api.Test;
+
+import spleetwaise.transaction.model.transaction.Category;
+import spleetwaise.transaction.testutil.TypicalTransactions;
+
+public class CategoryFilterPredicateTest {
+    private static final Category testCategory = DEFAULT_CATEGORY;
+
+    @Test
+    public void constructor_null_exceptionThrown() {
+        assertThrows(NullPointerException.class, () -> new CategoryFilterPredicate(null));
+    }
+
+    @Test
+    public void execute_category_success() {
+        CategoryFilterPredicate pred = new CategoryFilterPredicate(testCategory);
+
+        assertTrue(pred.test(TypicalTransactions.SEANOWESME));
+        assertFalse(pred.test(TypicalTransactions.DANIELDEBT));
+    }
+
+    @Test
+    public void execute_categoryDifferentCase_success() {
```

Comment:
We might not need this method here, as it handles the checking of string cases, which the `Category` class should already handle.